### PR TITLE
BRMS tutorial: add `rstan` example

### PR DESCRIPTION
**Pull request by falkmielke**
## Description
This PR adds an example of `rstan` usage with a raw Stan model to the existing `brms` tutorial.
I also rearranged that tutorial a bit (separate section for data).

I was able to test the `Rmd` code, all runs as intended. However, I did not knit or publish any of this to html, the structure and publication procedure of this repo seem too sophisticated for me (though great that it is based on hugo ssg!). I hope it will not crash GHA.

Thank you in advance for review and suggestions!

## Task list
- [X] My tutorial or article is placed in a subfolder of `tutorials/content`
- [X] The filename of my tutorial or article is `index.md`. In case of an Rmarkdown tutorial I have knitted my `index.Rmd` to `index.md` (both files are pushed to the repo). 
- [X] I have included `tags` in the YAML header (see the tags listed in the [tutorials website side bar](https://inbo.github.io/tutorials/) for tags that have been used before)
- [X] I have added `categories` to the YAML header and my category tags are from the [list of category tags](https://github.com/inbo/tutorials/blob/master/static/list_of_categories)
- [ ] I was not able to compile the tutorial into `html` files, and would appreciate if you original authors and tutorial maintainers could take over.

### CODE REVIEW

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd [hunk 1]
```diff
@@ -1046,10 +1049,186 @@ comp_waic %>%
...
-## Conclusie
+### Conclusie
 
 Zowel o.b.v. de PPC als vergelijkingen met verschillende model selectie criteria, kunnen we besluiten dat het tweede Poisson model met random intercepts het best past bij de data. In principe konden we dit ook verwachten op basis van onze eigen intuïtie en het design van de studie, nl. het gebruik van de Poisson distributie om aantallen te modelleren en het gebruik van random intercepts om te controleren voor een hiërarchisch design (habitats genest in sites).
 
+
+## Deep Dive: `rstan`
+### Stan: Wat? Waarom?!
+De `brms` package is een gemaksverpakking van `rstan`, dat zelf de functionaliteit van `stan` naar R brengt.
+Stan is een modeleringsomgeving, geschreven in de programmeertaal `C`, die de benodigdheden probabilistische ("Bayesiaanse") modellen implementeerd.
```
Comment (wlangera) [inline content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1060]:
```suggestion
Stan is een modeleringsomgeving, geschreven in de programmeertaal `C`, die de benodigdheden voor probabilistische ("Bayesiaanse") modellen implementeerd.
```

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd [hunk 2]
```diff
@@ -1046,10 +1049,186 @@ comp_waic %>%
...
+
+Meer info vind je [op het internet](https://mc-stan.org).
+
+Het voordeel van `brms` is bruikbaarheid: veel functies werken "out-of-the-box", met redelijke standaardwaarden en een syntaxis die lijkt op wat frequentisten gewend zijn.
```
Comment (wlangera) [inline content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1064]: niet enkel frequentisten, het is de gangbare syntax voor veel packages in R (zowel frequentistische als bayesiaanse) die met statistische modellen werken

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd [hunk 3]
```diff
@@ -1046,10 +1049,186 @@ comp_waic %>%
...
+Het relatieve gebruiksgemak gaat echter ten koste van de flexibiliteit en in zekere mate de leesbaarheid.
```
Comment (wlangera) [inline content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1065]:
```suggestion
Het relatieve gebruiksgemak kan echter ten koste gaan van de flexibiliteit en in zekere mate van de leesbaarheid.
```

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd [hunk 4]
```diff
@@ -1046,10 +1049,186 @@ comp_waic %>%
...
+
+Stan en `rstan` daarentegen leunen meer op de wiskundige formulering van modellen.
+Elk aspect van het model moet expliciet worden ingesteld, wat een voordeel kan zijn (bijvoorbeeld als u te maken hebt met niet-standaard use cases) of een nadeel (bijvoorbeeld als u modellen op niet-optionele manieren beveiligt).
```
Comment (wlangera) [inline content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1068]:
```suggestion
Elk aspect van het model moet expliciet worden ingesteld, wat een voordeel kan zijn (bijvoorbeeld als u te maken hebt met niet-standaard usecases) of een nadeel (bijvoorbeeld als u modellen op niet-optionele manieren beveiligt).
```

Comment (wlangera) [inline content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1068]: wat bedoel je met "een model beveiligen"?

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd [hunk 5]
```diff
@@ -1046,10 +1049,186 @@ comp_waic %>%
...
+
+
+```{r}
+library("rstan")
+conflicted::conflicts_prefer(rstan::extract)
+conflicted::conflicts_prefer(brms::loo)
+```
+
+### Model Definition
+The simple poisson model resembles [one of the `stan`dard examples](https://mc-stan.org/docs/stan-users-guide/posterior-prediction.html#posterior-prediction-for-regressions), which you can refer to for all further details and more.
```
Comment (wlangera) [inline content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1080]:
```suggestion
Het eenvoudige Poisson-model lijkt op [een van de `stan`daardvoorbeelden](https://mc-stan.org/docs/stan-users-guide/posterior-prediction.html#posterior-prediction-for-regressions), waarnaar u kunt verwijzen voor alle verdere details.
```

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd [hunk 6]
```diff
@@ -1046,10 +1049,186 @@ comp_waic %>%
...
+
+RMarkdown kan `stan`-chunks verwerken, hoewel de algemenere modeldefinitie is uitbesteed aan een apart "*.stan"-bestand.
+Alternatief kan je je model ook als grote tekstblok definiëren, zoals hieronder te zien.
```
Comment (wlangera) [inline content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1083]:
```suggestion
Alternatief kan je je model ook als een groot tekstblok definiëren, zoals hieronder te zien.
```

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd (hunk 5 above)
Comment (wlangera) [reply content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1080]: je herhaalt de zin ook verderop, lijn 1084

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd [hunk 7]
```diff
@@ -1046,10 +1049,186 @@ comp_waic %>%
...
+  model_name = "stan poisson model"
+  )
+```
+
+
+### Sampling
+
+Sampling doet vrijwel hetzelfde als hierboven, want in de kern is `brms` gewoon `stan`.
+
+```{r stan_sampling}
```
Comment (wlangera) [inline content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1134]: Ik zou hier ook zelf een burn-in gebruiken zoals we in de brms-voorbeelden hebben gedaan. Dan is het nog meer gelijkaardig als hierboven.

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd [hunk 8]
```diff
@@ -1046,10 +1049,186 @@ comp_waic %>%
...
+In het huidige geval is het voordeel van het overstappen op `stan` zeer beperkt.
+In andere gevallen kan het lonen.
+
+Weet dat Stan er voor u is, aarzel niet om de uitgebreide documentatie te raadplegen en wees niet bang om het te proberen!
+
+
+### Huiswerk: Hierarchical Model
+Om je modelleringsvaardigheden nog verder te ontwikkelen, kun je het "random intercept"-model implementeren en sampelen.
+
+In "Bayesiaanse" termen is het [algemene terminologie is "hiërarchisch" model](https://mc-stan.org/docs/stan-users-guide/regression.html#hierarchical-regression).
```
Comment (wlangera) [inline content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1169]:
```suggestion
In "Bayesiaanse" termen is de[algemene terminologie een "hiërarchisch" model](https://mc-stan.org/docs/stan-users-guide/regression.html#hierarchical-regression).
```

**Review by wlangera — CHANGES_REQUESTED** [review]
Hi Falk

Very nice addition.

I checked the Dutch version. I will assume the same comments apply for the English version.
General remarks:

- Could you keep the term burnin instead of warmup. I like burnin more.
- Any particular reason you provide the poisson_model.stan file in brms_eng folder and not in the brms_nl folder?
- I think I would like to put your section ("Deep Dive: rstan") in a separate section between "Vergelijken met frequentist statistics" and "Referenties". I think the use of rstan is quite advanced for the general users in INBO. Therefore, I think it is better that it is not in the middle of our tutorial, but behind it. People get a view of  the whole process of bayesian statistisc (in brms), and if they want (or if they heard about stan somewhere and want to look into it), they can progress further to your section. What do you think?

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd (hunk 7 above)
Comment (falkmielke) [reply content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1134]: Thank you for pointing this out, @wlangera !
I briefly asked @hansvancalster  and googled "burn-in tuning warmup bayesian stats". There is a [to-the-point discussion here](https://statmodeling.stat.columbia.edu/2017/12/15/burn-vs-warm-iterative-simulation-algorithms). My notion of changing came from the fact that we don't physically "burn" or "burn in" anything; thus I was surprised that apparently the term was considered appropriate for some MCMC algorithms. "Tuning", my actual favorite :metal:, is reserved for a certain sub-part of the warm-up phase.

Anyways, the experts seem to prefer "warm-up", and if you agree and confirm, I will in turn adjust this throughout the tutorial.

**Comment by falkmielke** [comment]
> From looking at the `content/tutorials/r_brms/index.md` file, I think that the html's should be rendered locally and commited to the repo [in this folder](https://github.com/inbo/tutorials/tree/master/static/html).

Thanks, I will try that, and we see where checks lead us.

**Comment by falkmielke** [comment]
> * Any particular reason you provide the poisson_model.stan file in brms_eng folder and not in the brms_nl folder?

Good point; I was thinking about git / data redundancy. 
Thinking one step further: 
- should they rather be in the `[...]/html` folder, close to what people actually read?
- ... or shouldn't they best be in a top level `/data` folder, or in `/inst`, to be generally accessible? 
Thank you for advice @wlangera @hansvancalster ! I will move the file and adjust links.

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd (hunk 4 above)
Comment (falkmielke) [reply content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd]: This was a translation fault of a faulty English sentence, which on top of that I overlooked in proofreading. My apologies! It is corrected.

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd (hunk 5 above)
Comment (falkmielke) [reply content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1080]: It was still in there from translating. thank you for the catch!

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd (hunk 2 above)
Comment (falkmielke) [reply content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1064]: Klopt! I generalized to "statisticians".

**Comment by falkmielke** [comment]
I incorporated the change suggestions and re-knitted the html reports. This also yielded two image-related issues, [here](https://github.com/inbo/tutorials/issues/348) and [here](https://github.com/inbo/tutorials/issues/347).

- the `cmdstanr` package is explicitly mentioned (suggestion by @hansvancalster )
- general code files are now in a general folder
- knit with low res images

Thank you for re-checking and publishing!

File: content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd (hunk 7 above)
Comment (wlangera) [reply content/tutorials/r_brms/brms_nl/workshop_1_mcmc_en_brms.Rmd:1267]: Ok that sounds good. Thanks

**Review by wlangera — APPROVED** [review]
Hi Falk, I have send you the images via mail.

Thanks for the changes

**Pull request merged**